source/common/network/connection_balancer_impl.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "envoy/network/connection_balancer.h"
+
+#include "absl/synchronization/mutex.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Implementation of connection balancer that does exact balancing. This means that a lock is held
+ * during balancing so that connection counts are nearly exactly balanced between handlers. This
+ * is "nearly" exact in the sense that a connection might close in parallel thus making the counts
+ * incorrect, but this should be rectified on the next accept. This balancer sacrifices accept
+ * throughput for accuracy and should be used when there are a small number of connections that
+ * rarely cycle (e.g., service mesh gRPC egress).
+ */
+class ExactConnectionBalancerImpl : public ConnectionBalancer {
```

Comment:
Is this balancer tested anywhere?

Reply:
Yeah it's tested everywhere as it's the NOP balancer installed by default.